python/cudf/benchmarks/API/bench_dataframe.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+"""Benchmarks of DataFrame methods."""
+
+import string
+
+import numpy
+import pytest
+from config import cudf, cupy
+from utils import accepts_cudf_fixture
+
+
+@pytest.mark.parametrize("N", [100, 1_000_000])
+def bench_construction(benchmark, N):
+    benchmark(cudf.DataFrame, {None: cupy.random.rand(N)})
+
+
+@accepts_cudf_fixture(cls="dataframe", dtype="float", cols=6)
+@pytest.mark.parametrize(
+    "expr", ["a+b", "a+b+c+d+e", "a / (sin(a) + cos(b)) * tanh(d*e*f)"]
+)
+def bench_eval_func(benchmark, expr, dataframe):
+    benchmark(dataframe.eval, expr)
+
+
+@accepts_cudf_fixture(
+    cls="dataframe", dtype="int", nulls=False, cols=6, name="df"
+)
+@pytest.mark.parametrize(
+    "nkey_cols",
+    [2, 3, 4],
+)
+def bench_merge(benchmark, df, nkey_cols):
+    benchmark(df.merge, df, on=list(df.columns[:nkey_cols]))
+
+
+# TODO: Some of these cases could be generalized to an IndexedFrame benchmark
+# instead of a DataFrame benchmark.
+@accepts_cudf_fixture(cls="dataframe", dtype="int")
+@pytest.mark.parametrize(
+    "values",
+    [
+        range(1000),
+        {f"key{i}": range(1000) for i in range(10)},
+        cudf.DataFrame({f"key{i}": range(1000) for i in range(10)}),
+        cudf.Series(range(1000)),
+    ],
+)
+def bench_isin(benchmark, dataframe, values):
+    benchmark(dataframe.isin, values)
+
+
+@pytest.fixture(
+    params=[0, numpy.random.RandomState, cupy.random.RandomState],
+    ids=["Seed", "NumpyRandomState", "CupyRandomState"],
```

Comment:
The performance does matter for a couple of reasons, mostly around the fact that our sampling implementation is different for sampling rows vs columns and you actually _can't_ use cupy arrays for column sampling (because it needs to happen on the host) but you could use either for sampling rows and there is a meaningful perf difference.